src/site/content/en/learn/css/specificity/index.md

```diff
@@ -0,0 +1,363 @@
+```
+
+### `!important` rule
+
+Lastly, an `!important` at the end of a CSS value gets a specificity score of **10,000 points**.
+This is the highest specificity that one individual item can get.
+
+An `!important` rule is applied to a CSS property,
+so everything in the overall rule (selector and properties) does not get the same specificity score.
+
+```css
+.my-class {
+	color: red !important; /* 10,000 points */
+	background: white; /* 10 points */
+}
+```
+
+@@TODO: self-assessment 1
+
+## Specificity in context
+
+The specificity of each selector that matches an element is added together.
+Consider this example HTML:
+
+```html
+<a class="my-class another-class" href="#">A link</a>
+```
+
+This link has two classes on it.
+Add the following CSS, and it gets **1 point of specificity**:
+
+```css
+a {
+  color: red;
+}
+```
+
+Reference one of the classes in this rule,
+it now has **11 points of specificity**:
+
+```css
+a.my-class {
+  color: green;
+}
+```
+
+Add the other class to the selector,
+it now has  **21 points of specificity**:
+
+```css
+a.my-class.another-class {
+  color: rebeccapurple;
+}
+```
+
+Add the `href` attribute to the selector,
+it now has **31 points of specificity**:
+
+```css
+a.my-class.another-class[href] {
+  color: goldenrod;
+}
+```
+
+Finally,add a `:hover` pseudo-class to all of that,
+the selector ends up with **41 points of specificity**:
+
+```css
+a.my-class.another-class[href]:hover {
+  color: lightgrey;
+}
+```
+
+@@TODO: self-assessment 2
+
+## Visualizing specificity
+
+In diagrams and specificity calculators,
+the specificity is often visualized like this:
+
+{% Img src="image/VbAJIREinuYvovrBzzvEyZOpw5w1/d1q5VKbFOQG2TBXoBtgm.png", alt="A diagram demonstrating most specific to least specific selectors", width="800", height="474" %}
+
+The left group is `id` selectors.
+The second group is class, attribute, and pseudo-class selectors.
+The final group is element and pseudo-element selectors.
+
+For reference, the following selector is `0-4-1`:
+
+```css
+a.my-class.another-class[href]:hover {
+  color: lightgrey;
+}
+```
+
+@@TODO: self-assessment 3
+
+## Pragmatically increasing specificity
+
+Let's say we have some CSS that looks like this:
+
+```css
+.my-button {
+	background: blue;
+}
+
+button[onclick] {
+	background: grey;
+}
+```
+
+With HTML that looks like this:
+
+```html
+<button class="my-button" onclick="alert('hello')">Click me</button>
+```
+
+<figure class="w-figure">
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'abpoxdR'
+} %}
+</figure>
+
+The button has a grey background,
+because the second selector earns **11 points of specificity** (`0-1-1`).
+This is because it has one type selector (`button`),
+which is **1 point** and an attribute selector (`[onclick]`), which is **10 points**.
+
+The previous rule—`.my-button`—gets **10 points** (`0-1-0`),
+because it has one class selector.
+
+If you want to give this rule a boost,
+repeat the class selector like this:
```

Comment:
```suggestion
a little hack you could do is to repeat the class selector like this:
```

Reply:
I'm not a fan of the term "hack" here, it's not a "hack", it's as per spec.

Reply:
The way it's written now in the content hierarchy makes it seem like it's common/the norm to write double classes like this. How can we call out that this isn't common behavior, though something to be aware of?